File: backend/hostel-management/src/main/java/com/hostel/hostel_management/service/MessService.java
package com.hostel.hostel_management.service;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

import com.hostel.hostel_management.entity.Fee;
import com.hostel.hostel_management.entity.MessAttendance;
import com.hostel.hostel_management.entity.MessMenu;



public interface MessService {
    MessMenu createMenu(MessMenu menu);
    List<MessMenu> getMenuByDate(LocalDate date);
    MessAttendance markAttendance(MessAttendance attendance);
    BigDecimal calculateMessBill(Long studentId);
    Fee generateMessFee(Long studentId);
}

File: frontend/hostel-frontend/src/services/messService.js
import API from "../api/axios";

export const createMenu = (data) =>
  API.post("/mess/menu", data);

export const getMenu = (date) =>
  API.get(`/mess/view-menu?date=${date}`);

export const markAttendance = (data) =>
  API.post("/mess/mess-attendance", data);

export const calculateBill = (studentId) =>
  API.get(`/mess/bill/${studentId}`);

export const generateMessFee = (studentId) =>
  API.post(`/mess/generate-fee/${studentId}`);